qcodes/dataset/plotting.py

```diff
@@ -23,6 +24,13 @@
 AxesTupleList = Tuple[List[matplotlib.axes.Axes],
                       List[Optional[matplotlib.colorbar.Colorbar]]]
 
+# list of kwargs for plotting function, so that kwargs can be passed to
+# :meth:`plot_by_id` and will be distributed to the respective plotting func.
+FIGURE_KWARGS = set(inspect.signature(plt.figure).parameters.keys())
+FIGURE_KWARGS.remove('kwargs')
+SUBPLOTS_KWARGS = set(inspect.signature(plt.subplots).parameters.keys())
+SUBPLOTS_KWARGS.remove('fig_kw')
+SUBPLOTS_KWARGS = FIGURE_KWARGS.union(SUBPLOTS_KWARGS)
```

Comment:
For readability i would prefer if the union was called something else that the individual set